txnsync/txngroups.go

```diff
@@ -60,23 +65,52 @@ func encodeTransactionGroups(inTxnGroups []transactions.SignedTxGroup) ([]byte,
 					stub.BitmaskGroup.SetBit(index)
 				}
 				if err := stub.deconstructSignedTransactions(index, &txn); err != nil {
-					return nil, fmt.Errorf("failed to encodeTransactionGroups: %w", err)
+					return nil, false, fmt.Errorf("failed to encodeTransactionGroups: %w", err)
 				}
 				index++
 			}
 		}
 	}
 	stub.finishDeconstructSignedTransactions()
 
-	return stub.MarshalMsg(protocol.GetEncodingBuf()[:0]), nil
+	encoded := stub.MarshalMsg(protocol.GetEncodingBuf()[:0])
+
+	if len(encoded) > 1000 && (1.0 / compressionSpeed) < (compressionSavings / float32(dataExchangeRate)) {
+		compressedBytes, err := compressTransactionGroupsBytes(encoded)
+		return compressedBytes, true, err
```

Comment:
in case you've failed to compress the encoded bytes, just fall back to return the encoded bytes without any compression. ( at this point, you might want to log a warning message as well )